lib/cli/node-modules-path.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var path  = require('path');
+
+module.exports = function(context) {
+  var nodePath = process.env.EMBER_NODE_PATH;
+  var contextPath = path.resolve(context);
+
+  if (nodePath) {
+    var nodeModulesPath = path.resolve(nodePath);
+
+    if (contextPath.indexOf(nodeModulesPath) === 0) {
+      return path.resolve(contextPath,'node_modules');
```

Comment:
@mixonic I think this is the correct behavior based on what the code currently does from my reading.  If `node_modules` path is `/Projects/node_modules` and a `contextPath` of `/Projects/node_modules/my-add-on` is passed in, we want to return `/Projects/node_modules/my-add-on/node_modules`

Right now this is reflected in the tests in this way.  Happy to break it out into an intention revealing function though.